ignite/engine/events.py

```diff
@@ -1,53 +1,71 @@
 
 from typing import Callable, Optional, Union, Any
 
-from enum import Enum
+from enum import Enum, EnumMeta
 import numbers
 import weakref
+from types import DynamicClassAttribute
 
 from ignite.engine.utils import _check_signature
 
-
 __all__ = [
     'Events',
     'State'
 ]
 
 
-class EventWithFilter:
-
-    def __init__(self, event: Any, filter: Callable):
-        if not callable(filter):
-            raise TypeError("Argument filter should be callable")
-        self.event = event
-        self.filter = filter
+class CallableEventWithFilter:
+    """Single Event containing a filter, specifying whether the event should
+    be run at the current event (if the event type is correct)
 
-    def __str__(self) -> str:
-        return "<%s event=%s, filter=%r>" % (self.__class__.__name__, self.event, self.filter)
+    Args:
+        value (str): The value (will only be set if not yet set by other parent class). Necessary for usage with
```

Comment:
This is a bit difficult to understand what user should pass here... please, rephrase it 
like "internal usage, do not touch"  ;)